app/services/cap_update_service.rb

```diff
@@ -0,0 +1,34 @@
+class CapUpdateService
+  attr_accessor :school
+
+  def initialize(args = {})
+    @school = args[:school]
+  end
+
+  def update!(order_state:, cap:)
+    update_order_state!(order_state)
+    allocation = update_cap!(cap)
+
+    notify_computacenter!(allocation.id)
+  end
+
+private
+
+  def update_order_state!(order_state)
+    @school.update!(order_state: order_state)
+  end
+
+  def update_cap!(cap)
+    allocation = SchoolDeviceAllocation.find_or_initialize_by(school_id: @school.id, device_type: 'std_device')
+    # we only take the cap from the user if they chose specific circumstances
+    # for both other states, we need to infer a new cap from the chosen state
+    allocation.cap = allocation.cap_implied_by_order_state(order_state: @school.order_state, given_cap: cap)
+    allocation.save!
+    allocation
+  end
+
+  def notify_computacenter!(allocation_id)
```

Comment:
Could we call this something else please? e.g. `update_caps_on_computacenter` or similar?

I initially confused this with sending a Notify email to Computacenter, which is a follow-up story that we will have to add very shortly to this service.

Reply:
Done in e004e03c